src/options.cpp

```diff
@@ -52,6 +53,49 @@ options_data::options_data()
     //   optionsdata.addme("OPTION_KEY_THAT_GETS_STRING_ENTRIES_ADDED_VIA_JSON", "thisvalue");
 }
 
+void options_data::enable_json(const std::string &lvar)
+{
+    post_json_verify[ lvar ] = std::string( 1, 001 ); // because "" might be valid
+}
+
+void options_data::add_retry(const std::string &lvar, const::std::string &lval)
+{
+    static const std::string blank_value( 1, 001 );
+    std::map<std::string, std::string>::const_iterator it = post_json_verify.find(lvar);
+    if ( it != post_json_verify.end() && it->second == blank_value ) {
+        // initialized with impossible value: valid
+        post_json_verify[ lvar ] = lval;
+    }
+}
+
+void options_data::add_value( const std::string &lvar, const std::string &lval,
+                              std::string lvalname )
+{
+    static const std::string blank_value( 1, 001 );
```

Comment:
Why exactly did you move the definitions of those `options_data` functions? They can be "used" once they have been declared (which happens in option.h, which is include at the very top of this file), the place of their definition does not matter at all. Have those functions actually been changed?